shell/utility/atom_content_utility_client.cc

```diff
@@ -139,6 +111,16 @@ bool AtomContentUtilityClient::HandleServiceRequest(
   return false;
 }
 
+void AtomContentUtilityClient::RunIOThreadService(
+    mojo::GenericPendingReceiver* receiver) {
+  if (auto factory_receiver =
+          receiver->As<proxy_resolver::mojom::ProxyResolverFactory>()) {
+    static base::NoDestructor<proxy_resolver::ProxyResolverFactoryImpl> factory(
```

Comment:
is this copied from somewhere? it seems kinda funky

Reply:
Yup, `ChromeContentUtilityClient::RunIOThreadService`, our proxy resolver service is managed exactly the way its done in chrome, so the code is identical.

> it seems kinda funky

Yeah I can see that at first,  but https://chromium-review.googlesource.com/c/chromium/src/+/1703095 and https://bugs.chromium.org/p/chromium/issues/detail?id=977637 details why it makes sense to be this way.